common/eventconsumption/eventindexmodel.hpp

```diff
@@ -91,6 +100,59 @@ class Storage
     friend bool operator < (const File& left, const char* right); // required to search a set by path
     friend bool operator < (const char* left, const File& right); // required to search a set by path
 
+    class PageCache
+    {
+    public:
+        static constexpr __uint64 DefaultPageSize = 8192; // 8K page size
+        class Key
+        {
+        public:
+            __uint64 fileId{0};
+            __uint64 offset{0};
+            Key(__uint64 _fileId, __uint64 _offset) : fileId(_fileId), offset(_offset) {}
+            bool operator == (const Key& other) const { return fileId == other.fileId && offset == other.offset; }
+        };
+        class KeyEqual
+        {
+        public:
+            bool operator () (const Key* left, const Key* right) const
+            {
+                return ((left == right) || (left->fileId == right->fileId && left->offset == right->offset));
+            }
+        };
+        class KeyHash
+        {
+        public:
+            std::size_t operator()(const Key* key) const
+            {
+                struct { __uint64 f; __uint64 s; } buf{key->fileId, key->offset};
+                return hashc_fnv1a((byte*)&buf, sizeof(buf), fnvInitialHash32);
+            }
+        };
+        class Value : public Key
```

Comment:
Add a comment:
// Value needs to contain the key so that when the LR element is removed from the list it can be resolved in the hash table.